app/lib/getLatestPeriod.js

```diff
@@ -0,0 +1,20 @@
+const log = require('./logger');
+const utils = require('./utils');
+
+function getLatestPeriod(periods) {
+  return new Promise((resolve, reject) => {
+    try {
+      log.time('Getting latest period took');
+      const latestPeriod = utils.getMostRecentPeriod(periods);
+      log.timeEnd('Getting latest period took');
+      log.info(`Latest period is ${latestPeriod}`);
+
+      resolve(latestPeriod);
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+module.exports = getLatestPeriod;
+
```

Comment:
As this method is called from a previous promise it is auto-magically wrapped in a promise so a new Promise isn't required. The body could just be
```
  log.time('Getting latest period took');
  const latestPeriod = utils.getMostRecentPeriod(periods);
  log.timeEnd('Getting latest period took');
  log.info(`Latest period is ${latestPeriod}`);
  return latestPeriod;
```